todo-app/src/TodoList.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import Todo from './Todo';
+import initialTodoData from './data/todo-data.json';
+
+
+
+export default class TodoList extends React.Component {
+
+  state = {
+    todos: initialTodoData,
+  }
+  //done checkbox event handler
+  handleDoneToggle = (todoId) => {
+    this.setState({
+      todos: this.state.todos.map(todo => {
+        if (todo.id === todoId) {
+          todo.done = !todo.done
```

Comment:
In this line, you're mutating (modifying) the todo item. In React, when using setState, make sure to make your code immutable. the `map` function is an example of an immutable function. Can you modify this code to make it fully immutable?